segv when searching in kv search after reading-in Haybale file

Haystack is panicing with a runtime error when attempting to use -kv search after reading in a Haybale file.

Firstly, the success case when importing JSON file and using kv search:
```
dkaiser@baler:~/code/haystack$ go run cmd/haystack.go -i testdata/head5.json -kv flow_id 2095840221535198
Haystack - Haystack log management system test & benchmark tool
Copyright (C) 2023 Arjen Lentz & Lentz Pty Ltd; All Rights Reserved
Licenced under the Affero General Public Licence (AGPL) v3(+)

Ingesting file 'testdata/head5.json'
Inserted 5 JSON lines, duration: 998.686µs
Sorting all (1) haybale(s)...
Running the Go garbage collector
Haybale sort & de-dup
num_haystalks = 103, Heap was=1M, now=1M (0.00% gain), TotalAlloc=1M
Haybale sort duration: 10.622069ms
Searching for key flow_id = 2095840221535198
Looking in Haybale 0 (103 stalks)
{"_timestamp":"2023-06-04T00:01:03.766246+0000","app_proto":"tls","dest_ip":"192.168.179.57","dest_port":"443","event_type":"flow","flow.age":"9","flow.alerted":"false","flow.bytes_toclient":"59957","flow.bytes_toserver":"38099","flow.end":"2023-06-03T23:59:46.805686+0000","flow.pkts_toclient":"114","flow.pkts_toserver":"116","flow.reason":"timeout","flow.start":"2023-06-03T23:59:37.924638+0000","flow.state":"closed","flow_id":"2095840221535198","in_iface":"eth2","proto":"TCP","src_ip":"160.101.179.62","src_port":"48050","tcp.ack":"true","tcp.fin":"true","tcp.psh":"true","tcp.rst":"true","tcp.state":"closed","tcp.syn":"true","tcp.tcp_flags":"1f","tcp.tcp_flags_tc":"1b","tcp.tcp_flags_ts":"1e"}
1 matches, duration: 112.466µs
```

Now, importing JSON and writing of a Haybale file:
```
dkaiser@baler:~/code/haystack$ go run cmd/haystack.go -i testdata/head5.json -w head5.hay
Haystack - Haystack log management system test & benchmark tool
Copyright (C) 2023 Arjen Lentz & Lentz Pty Ltd; All Rights Reserved
Licenced under the Affero General Public Licence (AGPL) v3(+)

Ingesting file 'testdata/head5.json'
Inserted 5 JSON lines, duration: 534.282µs
Writing Haystack file 'head5.hay'
Running the Go garbage collector
Haybale sort & de-dup
num_haystalks = 103, Heap was=1M, now=1M (0.00% gain), TotalAlloc=1M

dkaiser@baler:~/code/haystack$ ls -la head5.hay 
-rw------- 1 dkaiser dkaiser 2836 Jun 30 17:14 head5.hay
```

And... the error, when reading the Haybale file and searching:  (same key value as success search above after importing JSON)
```
dkaiser@baler:~/code/haystack$ go run cmd/haystack.go -r head5.hay -kv flow_id 2095840221535198
Haystack - Haystack log management system test & benchmark tool
Copyright (C) 2023 Arjen Lentz & Lentz Pty Ltd; All Rights Reserved
Licenced under the Affero General Public Licence (AGPL) v3(+)

Reading Haystack file 'head5.hay'
Sorting all (1) haybale(s)...
Haybale sort duration: 10.58µs
Searching for key flow_id = 2095840221535198
Looking in Haybale 0 (103 stalks)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4e013f]

goroutine 1 [running]:
openacta.dev/haystack.(*Haystack).SearchKeyVal(0x6012a0, {0x7fff79af80dd, 0x7}, {0x7fff79af80e5, 0x10})
	/home/dkaiser/code/haystack/mem_search.go:144 +0x99f
main.main()
	/home/dkaiser/code/haystack/cmd/haystack.go:117 +0x2a5
exit status 2
```

Note: I'm not 100% sure I've invoked Haystack properly, but regardless am getting a segv with the -kv command, so am reporting it.

```
dkaiser@baler:~/code/haystack$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"

dkaiser@baler:~/code/haystack$ go version
go version go1.18.1 linux/amd64
```

--- a/disk2mem.go
+++ b/disk2mem.go
@@ -204,7 +204,7 @@ func (p *Haystack) getDisk2MemDictionary(content []byte, first_dictionary bool) 
 		return fmt.Errorf("read num dkeys %d > %d possible", read_num_dkeys, max_dkeys)
 	}
 
-	for i := 1; i < read_num_dkeys; i++ {
+	for i := 0; i < read_num_dkeys; i++ {
 		dkey, key := getKeyFromData(reader)
 
 		//fmt.Fprintf(os.Stderr, "dkey[%d]=%-10s\r", dkey, *key) // DEBUG

